src/iuse_actor.cpp

```diff
@@ -917,3 +917,619 @@ bool extended_firestarter_actor::can_use( const player* p, const item* it, bool
+        int maxMalus = -250 * (5.0 / (float)p->skillLevel("survival"));
+        int duration = 300 * (5.0 / (float)p->skillLevel("survival"));
+        int decayDelay = 30 * (5.0 / (float)p->skillLevel("survival"));
+
+        if (p->has_trait("PACIFIST")) {
+            moraleMalus *= 5;
+            maxMalus *= 3;
+        } else if (p->has_trait("PRED1")) {
+            moraleMalus /= 4;
+        } else if (p->has_trait("PRED2")) {
+            moraleMalus /= 5;
+        }
+
+        p->add_morale(MORALE_MUTILATE_CORPSE, moraleMalus, maxMalus, duration, decayDelay);
+    }
+
+    const int selected_corpse = amenu.ret - 1;
+
+    const item *body = corpses[selected_corpse];
+    const mtype *mt = body->get_mtype();
+
+    // HP range for zombies is roughly 36 to 120, with the really big ones having 180 and 480 hp.
+    // Speed range is 20 - 120 (for humanoids, dogs get way faster)
+    // This gives us a difficulty ranging rougly from 10 - 40, with up to +25 for corpse damage.
+    // An average zombie with an undamaged corpse is 0 + 8 + 14 = 22.
+    int difficulty = (body->damage * 5) + (mt->hp / 10) + (mt->speed / 5);
+    // 0 - 30
+    int skills = p->skillLevel("survival") + p->skillLevel("firstaid") + (p->dex_cur / 2);
+    skills *= 2;
+
+    int success = rng(0, skills) - rng(0, difficulty);
+
+    const int moves = difficulty * 1200 / p->skillLevel("firstaid");
+
+    p->assign_activity(ACT_MAKE_ZLAVE, moves);
+    p->activity.values.push_back(success);
+    p->activity.str_values.push_back(corpses[selected_corpse]->display_name());
+    return it->type->charges_to_use();
+}
+
+bool enzlave_actor::can_use( const player *p, const item*, bool, const point& ) const
+{
+    return p->get_skill_level( "survival" ) > 1 && p->get_skill_level( "firstaid" ) > 1;
+}
+
+void fireweapon_off_actor::load( JsonObject &obj )
+{
+    success_message     = obj.get_string( "success_message" );
+    lacks_fuel_message  = obj.get_string( "lacks_fuel_message" );
+    failure_message     = obj.get_string( "failure_message", "" );
+    noise               = obj.get_int( "noise", 0 );
+    moves               = obj.get_int( "moves", 0 );
+    success_chance      = obj.get_int( "success_chance", INT_MIN );
+}
+
+iuse_actor *fireweapon_off_actor::clone() const
+{
+    return new fireweapon_off_actor( *this );
+}
+
+long fireweapon_off_actor::use( player *p, item *it, bool t, point ) const
+{
+    if( t ) {
+        return 0;
+    }
+
+    p->moves -= moves;
+    if( rng( 0, 10 ) - it->damage > success_chance && 
+          it->charges > 0 && !p->is_underwater() ) {
+        if( noise > 0 ) {
+            sounds::sound( p->posx(), p->posy(), noise, _(success_message.c_str()) );
+        } else {
+            p->add_msg_if_player( _(success_message.c_str()) );
+        }
+
+        it->make( target_id );
+        it->active = true;
+    } else {
+        p->add_msg_if_player( m_bad, _(failure_message.c_str()) );
```

Comment:
Careful here: `failure_message` defaults to an empty string and the translation system will translate the empty string into some debug information about the translation data, you probably don't want that. Check whether the message is empty before displaying it. You can see the translation of the empty string in #10882.